components/raftstore/src/store/entry_storage.rs

```diff
@@ -550,11 +546,8 @@ pub fn init_applied_term<ER: RaftEngine>(
 
     match raft_engine.get_entry(region.get_id(), apply_state.applied_index)? {
         Some(e) => Ok(e.term),
-        None => Err(box_err!(
-            "[region {}] entry at apply index {} doesn't exist, may lose data.",
-            region.get_id(),
-            apply_state.applied_index
-        )),
+        // None means apply is ahead of persist, in this case the raft term must not be changed.
+        None => Ok(apply_state.commit_term),
```

Comment:
How about
1. judge if `applied_index` is greater than the last index
2. if true, return commit_term, otherwise maintain the original code path